bundles/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/util/SCPDUtil.java

```diff
@@ -141,7 +144,7 @@ public SCPDUtil(HttpClient httpClient, String endpoint) throws SCPDException {
      * @param serviceId the service id
      * @return the service
      */
-    public Optional<SCPDScpdType> getService(String serviceId) {
+    public Optional<@Nullable SCPDScpdType> getService(String serviceId) {
```

Comment:
This is wrong. The content of the `Optional` is always non-null. If no value is present, the `Optional` is empty. Annotating `@Nullable` says "if you `get` the value, it can be `null`". The main reson why you need to add it here is that the null-annotations don't support `Optional.ofNullable`.

Reply:
Same question here: How can we solve it if the `SuppressWarnings` does not work? Isn't the annotation an acceptable workaround? After all, we before had null as a possible value everywhere and if we now have it at a few places more than we are able to specify, it is imho still ok. That's the side effect of setting "NonNullByDefault" everywhere...